plugins/analysis/dashboard/ws.go

```diff
@@ -100,20 +111,57 @@
 	clientID, wsClient := registerWSClient()
 	defer removeWsClient(clientID)
 
+	manaDashboardHostAddress := config.Node().GetString(CfgManaDashboardAddress)
+	registerReadHandlers(dashboard.MsgRequestManaDashboardAddress, func(_ interface{}) {
+		msg := &wsmsg{
+			Type: dashboard.MsgManaDashboardAddress,
+			Data: manaDashboardHostAddress,
+		}
+		broadcastWsMessage(msg)
+	})
+
 	// replay autopeering events from the past upon connecting a new client
 	analysisserver.ReplayAutopeeringEvents(createAutopeeringEventHandlers(ws))
 
 	// replay FPC past events
 	replayFPCRecords(ws)
 
 	for {
-		msg := <-wsClient.channel
-		if err := ws.WriteJSON(msg); err != nil {
-			break
-		}
-		if err := ws.SetWriteDeadline(time.Now().Add(webSocketWriteTimeout)); err != nil {
-			break
+		select {
+		case msg := <-wsClient.channel:
+			if err := ws.WriteJSON(msg); err != nil {
+				break
+			}
+			if err := ws.SetWriteDeadline(time.Now().Add(webSocketWriteTimeout)); err != nil {
+				break
+			}
+		case <-shutdownSignal:
+			return nil
+		default:
+			_, msg, err := ws.ReadMessage()
```

Comment:
is this blocking if there are no incoming msgs?

Reply:
no